jupyter_sphinx/execute.py

```diff
@@ -149,7 +149,7 @@ def apply(self):
 
             notebook = execute_cells(
                 kernel_name,
-                [nbformat.v4.new_code_cell(node.astext()) for node in nodes],
+                [nbformat.v4.new_code_cell(node.astext() if node["execute"] else "") for node in nodes],
```

Comment:
This will have a downside of the cells that aren't executed not available for download in a generated notebook or a script. Should we improve on that by:

- Remembering at which locations these cells would be inserted.
- Reinserting them into the notebook after the execution has finished (and also handling the outputs at that point)

Reply:
If such a thing existed, we could mark the cells with "please don't execute tag", but it doesn't exist.

Reply:
This is already done - see the other change made to execute.py. The insertion of an empty cell here means that zipping the nodes and cells of the notebook still keeps them aligned correctly and we can hence replace the empty cells with the given input/output code as needed

Reply:
Ah, great, sorry that I didn't catch that first! Please leave a comment here about the purpose of the empty code cells.

I was also for a second concerned that this would cause a hit in performance due to pinging kernel with empty messages, but it seems nbclient already skips executing empty code cells.